lib/ansible/modules/database/postgresql/postgresql_pg_hba.py

```diff
@@ -0,0 +1,673 @@
+        Therefore for ipv4, we use prefixlen (0-32) * 4 for weight,
+        which corresponds to ipv6 (0-128).
+        '''
+        if order not in PG_HBA_ORDERS:
+            raise PgHbaRuleError('{0} is not a valid order'.format(order))
+
+        if self['type'] == 'local':
+            sourceobj = ''
+            # local is always 'this server' and therefore considered /32
+            srcweight = 130  # (Sort local on top of all)
+        else:
+            sourceobj = self.source()
+            if isinstance(sourceobj, ipaddress.IPv4Network):
+                srcweight = sourceobj.prefixlen * 4
+            elif isinstance(sourceobj, ipaddress.IPv6Network):
+                srcweight = sourceobj.prefixlen
+            elif isinstance(sourceobj, str):
+                # You can also write all to match any IP address,
+                # samehost to match any of the server's own IP addresses,
+                # or samenet to match any address in any subnet that the server is connected to.
+                if sourceobj == 'all':
+                    # (all is considered the full range of all ips, which has a weight of 0)
+                    srcweight = 0
+                elif sourceobj == 'samehost':
+                    # (sort samehost second after local)
+                    srcweight = 129
+                elif sourceobj == 'samenet':
+                    # Might write some fancy code to determine all prefix's
+                    # from all interfaces and find a sane value for this one.
+                    # For now, let's assume IPv4/24 or IPv6/96 (both have weight 96).
+                    srcweight = 96
+                elif sourceobj[0] == '.':
+                    # suffix matching (domain name), let's asume a very large scale
+                    # and therefore a very low weight IPv4/16 or IPv6/64 (both have weight 64).
+                    srcweight = 64
+                else:
+                    # hostname, let's asume only one host matches, which is
+                    # IPv4/32 or IPv6/128 (both have weight 128)
+                    srcweight = 128
+
+        if self['db'] == 'all':
+            dbweight = numdbs
+        elif self['db'] == 'replication':
+            dbweight = 0
+        elif self['db'] in ['samerole', 'samegroup']:
+            dbweight = 1
+        else:
+            dbweight = 1 + self['db'].count(',')
+
+        if self['usr'] == 'all':
+            uweight = numusers
+        else:
+            uweight = 1
+
+        ret = []
+        for character in order:
+            if character == 'u':
+                ret.append(uweight)
+            elif character == 's':
+                ret.append(srcweight)
+            elif character == 'd':
+                ret.append(dbweight)
+        ret.append(sourceobj)
+
+        return tuple(ret)
+
+
+def main():
+    '''
+    This function is the main function of this module
+    '''
+    module = AnsibleModule(
```

Comment:
We're changing this construction to using pg_utils from postgres lib like this [#52077](https://github.com/ansible/ansible/pull/52077/files):
```
    argument_spec = postgres_common_argument_spec()
    argument_spec.update(
        table=dict(type='str', required=True, aliases=['name']),
        state=dict(type='str', default="present", choices=["absent", "present"]),
        db=dict(type='str', default='', aliases=['login_db']),
        port=dict(type='int', default=5432, aliases=['login_port']),
        ssl_mode=dict(type='str', default='prefer', choices=['allow', 'disable', 'prefer', 'require', 'verify-ca', 'verify-full']),
        ssl_rootcert=dict(type='str'),
        tablespace=dict(type='str'),
        owner=dict(type='str'),
        unlogged=dict(type='bool'),
        like=dict(type='str'),
        including=dict(type='str'),
        rename=dict(type='str'),
        truncate=dict(type='bool'),
        columns=dict(type='list'),
        storage_params=dict(type='list'),
        session_role=dict(type='str'),
    )
    module = AnsibleModule(
        argument_spec=argument_spec,
        supports_check_mode=True,
    )
```

Reply:
On this comment: I tried this, but it breaks, since postgresql_pg_hba does not have all of the postgres stuff to login, which all others do. We want to change that, but that is not in yet.
So I reverted this for now (commented the lines), and will add that when I implement the 'login and check' feature. Not sure if that will be in in this commit, bit that is fine. It can be added later.

Reply:
Ok